### PR TITLE
Support Bevy 0.11

### PR DESCRIPTION
**Pull request by bardt**
Most changes were quite straightforward. 

The most confusion came from the changes in window coordinates in some cases: https://bevyengine.org/learn/migration-guides/0.10-0.11/#consistent-screen-space-coordinates It made some previously required adjustments redundant. 

This depends on release of `bevy_eventlistener` with Bevy 0.11 support. In the meantime, I depend on a specific revision.

I also encountered a problem in `gltf` example due to https://github.com/aevyrie/bevy_mod_raycast/issues/86

Features "ktx2", "zstd", and "tonemapping_luts" turned out to be required for 3d examples to render properly. I guess it's connected to the new default tone mapper.

### CODE REVIEW

**Comment by bardt** [comment]
After I finished working in this one, I noticed that similar work was already done in https://github.com/aevyrie/bevy_mod_picking/pull/231 However, I failed to compile the branch locally, and by looking at the diff I doubt that all examples work as intended (I expect some of them to display debug data in the wrong vertical position etc.). I think it is worth checking my version of changes for the fixes.

**Comment by ostwilkens** [comment]
This PR is the first version of the plugin that I've used, so I don't know if this issue is related to 0.11 or not. The other 0.11 PR has the same issue: touch events for web builds behave as if the Y axis is flipped. 
Here is a build where this occurs: https://ostwilkens.github.io/230731/

**Comment by bardt** [comment]
@ostwilkens could you check again? I think it should be fixed now, but don't have a touch device at hand to test.

**Comment by ostwilkens** [comment]
@bardt That did it, thank you! It can be tested with chrome dev tools, but I tested on iOS Safari as well.

**Comment by 772** [comment]
These changes work perfectly fine for my bevy 0.11 project.

**Comment by AllenDang** [comment]
Any update? Waiting for it to upgrade my project to bevy 0.11...

**Comment by ostwilkens** [comment]
> Any update? Waiting for it to upgrade my project to bevy 0.11...

No need to wait! :-)
`bevy_mod_picking = { version = "0.14.0", git = "https://github.com/bardt/bevy_mod_picking.git", rev = "7ecb60e08735ce82f9ff03ba5adb434c1e336afd" }`

**Comment by bardt** [comment]
I'm looking forward to @aevyrie finding the time soon to review both this pull request and the one in `bevy_mod_raycast`. However, I haven't heard from them yet, so I'm not sure when this will happen.

**Comment by aevyrie** [comment]
This is on my todo list tonight. :)

**Review by aevyrie — APPROVED** [review]
Looks great, thank you.

Running through the examples, trying to debug some strange stuttering behavior I haven't seen while updating any other crates.

File: crates/bevy_picking_core/src/pointer.rs [hunk 1]
```diff
@@ -279,8 +280,9 @@ impl Location {
 
         camera
             .logical_viewport_rect()
-            .map(|(min, max)| {
-                (position - min).min_element() >= 0.0 && (position - max).max_element() <= 0.0
+            .map(|rect| {
+                (position - rect.min).min_element() >= 0.0
+                    && (position - rect.max).max_element() <= 0.0
```
Comment (Vrixyz) [inline crates/bevy_picking_core/src/pointer.rs:285]: an alternative:

```suggestion
           .map(|Rect { min, max }| {
                (position - min).min_element() >= 0.0 && (position - max).max_element() <= 0.0
```

**Comment by aevyrie** [comment]
I'm seeing an issue with split screen. It appears the passed in coordinates are offset. It looks to me as if the viewport to world function is not accounting for the viewport origin, and instead using the window coordinates.

https://github.com/aevyrie/bevy_mod_picking/assets/2632925/4fb3f238-3759-4b53-bf78-2fd5263891a3

**Pull request merged**